### PR TITLE
Updated CodeQL Scan :) 

### PR DESCRIPTION
**Pull request by NickLiffen**
Hey 👋 

I hope this PR finds you well :) 

I'm one of the architects at GitHub on the GitHub Advanced Security Team 👋 

Great work on this project :) I wondered how open you would be to merge this PR?

The main change is introducing this line:

```
        queries: security-extended
```

I made a few other small semantically changes :) The workflow should run as is now, just with a different query suite :) 

Would love your feedback :) Great work 💯

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
+      security-events: write
     strategy:
       fail-fast: false
-    runs-on: ubuntu-latest
+      matrix:
+        language: [ 'javascript' ]
     steps:
     - name: Checkout repository
-      uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f #v2: v2.3.4 available
+      uses: actions/checkout@v2
```
Comment (bkimminich) [inline .github/workflows/codeql-analysis.yml]: Could you pin the checkout-action back to the commit hash, please? See https://github.com/juice-shop/juice-shop/pull/1687.

Comment (NickLiffen) [reply .github/workflows/codeql-analysis.yml]: Hey @bkimminich 👋 of course I can :) I wondered why you done that but it makes sense now :)

Comment (NickLiffen) [reply .github/workflows/codeql-analysis.yml]: Thanks for the feedback :) Done @bkimminich 👍

**Comment by NickLiffen** [comment]
@bkimminich when is the next deployment 🚋 leaving the station from `dev` to `main` 😄 don't need exact dates just curious :)

**Comment by bkimminich** [comment]
Thanks a lot! I'll go over the findings and close all intended vulns CodeQL might now have found as "Won't Fix" later!

As this is your first contribution to this project, our "stickers!!!"-policy applies. If you'd like me to send you some Juice Shop sickers and postcards, just let me know where to! (Also if I should make the package a bit bigger so you can give your colleagues some as well... 😁)

**Comment by bkimminich** [comment]
> @bkimminich when is the next deployment 🚋 leaving the station from `dev` to `main` 😄 don't need exact dates just curious :)

Some time next week probably, maybe already over this weekend.

**Comment by NickLiffen** [comment]
> Some time next week probably, maybe already over this weekend.

Okay amazing 👍 may look to get one more PR in before then (another small one) 

Thanks for your help :)

**Pull request merged**